prody/proteins/interactions.py

```diff
@@ -0,0 +1,3563 @@
+       if i not in ls:
+           ls.append(i)
+           newList.append(list_of_interactions[no])
+    return newList
+
+
+def get_permutation_from_dic(dictionary, key):
+    """Check permutations of residue pairs in a dictionary
+    format: key=('VAL8', 'LEU89')
+    ('VAL8', 'LEU89') and ('LEU89', 'VAL8') will be check and corresponding
+    value will be returned."""
+    
+    if key in dictionary:
+        return dictionary[key]
+
+    reversed_key = tuple(reversed(key))
+    if reversed_key in dictionary:
+        return dictionary[reversed_key]
+
+    return 0
+
+
+def filterInteractions(list_of_interactions, atoms, **kwargs):
+    """Return interactions based on selection."""
+    
+    if 'selection' in kwargs:
+        if 'selection2' in kwargs:
+            if not 'chid' in kwargs['selection'] and not 'chain' in kwargs['selection']:
+                LOGGER.warn('selection does not include chid or chain, so no filtering is performed')
+                return list_of_interactions
+
+            if not 'chid' in kwargs['selection2'] and not 'chain' in kwargs['selection2']:
+                LOGGER.warn('selection2 does not include chid or chain, so no filtering is performed')
+                return list_of_interactions
+
+            ch1 = kwargs['selection'].split()[-1] 
+            ch2 = kwargs['selection2'].split()[-1] 
+            final = [i for i in list_of_interactions if (i[2] == ch1 and i[5] == ch2) or (i[5] == ch1 and i[2] == ch2)]
+        else:
+            p = atoms.select('same residue as protein within 10 of ('+kwargs['selection']+')')
+            if p is None:
+                LOGGER.warn('selection did not work, so no filtering is performed')
+                return list_of_interactions
+
+            x = p.select(kwargs['selection']).getResnames()
+            y = p.select(kwargs['selection']).getResnums()
+            listOfselection = np.unique(list(map(lambda x, y: x + str(y), x, y)))
+            final = [i for i in list_of_interactions if i[0] in listOfselection or i[3] in listOfselection]
+    elif 'selection2' in kwargs:
+        LOGGER.warn('selection2 by itself is ignored')
+    else:
+        final = list_of_interactions
+    return final
+
+
+def calcHydrophobicOverlapingAreas(atoms, **kwargs):
+    """Provide information about hydrophobic contacts between pairs of residues based on 
+    the regsurf program.
+
+    :arg atoms: an Atomic object from which residues are selected
+    :type atoms: :class:`.Atomic`
+    
+    :arg selection: selection string of hydrophobic residues
+    :type selection: str
+    
+    :arg hpb_cutoff: cutoff for hydrophobic overlaping area values
+        by default is 0.0
+    :type hpb_cutoff: float or int
+    
+    :arg cumulative_values: sum of results for pairs of residues or
+        for single residues without distinguishing pairs,
+        by default is None
+    :type cumulative_values: 'pairs' or 'single_residues'    
```

Comment:
this should be a proper type. Is it a string or a list or what? You can describe what's inside it in the part above